include/seqan/index/find2_index_approx.h

```diff
@@ -39,138 +39,184 @@
+    std::array<uint8_t, N> u; // maximum number of errors at the end of the corresponding block
 
-    std::vector<uint8_t> blocklength; // cumulated values / prefix sums
-    uint16_t startPos;
+    std::array<uint32_t, N> blocklength { }; // cumulated values / prefix sums
+    uint32_t startPos = 0;
     // first character of the needle starts with position 1 (not with 0)
     // if initialDirection is true, startPos is one character to the left of the block that is searched first
     // otherwise, startPos is one character right from the block that is searched first
 
-    bool initialDirection; // true <=> goToRight
-} Search;
-
-typedef std::vector<Search> SearchScheme;
-
-// SearchScheme FAIL
-SearchScheme schemeFAIL
-{
-};
-
-// SearchScheme with 0 errors
-SearchScheme scheme_0_0
-{
-    { {1}, {0}, {0}, {0}, 0, false }
-};
-
-// SearchScheme with at most 1 error
-SearchScheme scheme_0_1
-{
-    { {1, 2}, {0, 0}, {0, 1}, {0, 0}, 0, false },
-    { {2, 1}, {0, 1}, {0, 1}, {0, 0}, 0, false }
-};
-
-// SearchScheme with at most 2 errors
-SearchScheme scheme_0_2
-{
-    { {2, 1, 3, 4}, {0, 0, 1, 1}, {0, 0, 2, 2}, {0, 0, 0, 0}, 0, false },
-    { {3, 2, 1, 4}, {0, 0, 0, 0}, {0, 1, 1, 2}, {0, 0, 0, 0}, 0, false },
-    { {4, 3, 2, 1}, {0, 0, 0, 2}, {0, 1, 2, 2}, {0, 0, 0, 0}, 0, false }
+    bool initialDirection = false; // true <=> goToRight
 };
 
-// SearchScheme with at most 3 errors
-SearchScheme scheme_0_3
-{
-    { {1, 2, 3, 4, 5}, {0, 0, 0, 2, 2}, {0, 0, 3, 3, 3}, {0, 0, 0, 0, 0}, 0, false },
-    { {4, 3, 2, 1, 5}, {0, 0, 0, 0, 0}, {1, 1, 2, 2, 3}, {0, 0, 0, 0, 0}, 0, false },
-    { {5, 4, 3, 2, 1}, {0, 0, 0, 0, 3}, {0, 2, 2, 3, 3}, {0, 0, 0, 0, 0}, 0, false }
-};
-
-// SearchScheme with at most 4 errors
-SearchScheme scheme_0_4
-{
-    { {1, 2, 3}, {0, 0, 3}, {1, 4, 5}, {0, 0, 0}, 0, false },
-    { {2, 3, 1}, {0, 0, 0}, {2, 3, 5}, {0, 0, 0}, 0, false },
-    { {3, 2, 1}, {0, 3, 5}, {0, 5, 5}, {0, 0, 0}, 0, false }
-};
-
-// SearchScheme with 1-1 errors
-SearchScheme scheme_1_1
-{
-    { {1, 2}, {0, 1}, {0, 1}, {0, 0}, 0, false },
-    { {2, 1}, {0, 1}, {0, 1}, {0, 0}, 0, false }
-};
-
-SearchScheme scheme_1_2 // TODO
-{
-    { {1, 2, 3}, {0, 0, 1}, {0, 2, 2}, {0, 0, 0}, 0, false },
-    { {3, 2, 1}, {0, 0, 2}, {0, 1, 2}, {0, 0, 0}, 0, false },
-    { {2, 1, 3}, {0, 1, 1}, {0, 1, 2}, {0, 0, 0}, 0, false }
-};
+template <size_t min, size_t max, typename TVoidType = void>
+struct SearchSchemes;
+// {
+//     static_assert(false, "The error configuration is invalid or there exists no such search scheme.");
+// };
 
-SearchScheme scheme_1_3 // TODO
+template <typename TVoidType>
+struct SearchSchemes<0, 0, TVoidType>
 {
-
+    static constexpr std::array<Search<1>, 1> VALUE { {{ {{1}}, {{0}}, {{0}} }} };
 };
 
-SearchScheme scheme_1_4 // TODO
-{
-
-};
+template <typename TVoidType>
+constexpr std::array<Search<1>, 1> SearchSchemes<0, 0, TVoidType>::VALUE;
```

Comment:
I think you might be able to get rid of all the declerations and replace them with one generic one:

```cpp
template <typename TVoidType>
struct SearchSchemes<0, 0, TVoidType>
{
   using TYPE = std::array<Search<1>, 1>;
   static constexpr TYPE VALUE { {{ {{1}}, {{0}}, {{0}} }} };
};
``` 
and then once for all:
```cpp
template <size_t min, size_t max, typename TVoidType = void>
constexpr typename SearchSchemes<min, max, TVoidType>::TYPE SearchSchemes<min, max, TVoidType>::VALUE;
```
You could even get rid of the VoidType like this, but I am not sure if that then means, that the specialised classes are included again 🤔